### PR TITLE
Fix(evm): no genesis change after evm change, no matter protocol-feature-evm enabled or not

### PR DESCRIPTION
**Pull request by ailisp**
Protect genesis change behind protocol_feature_evm feature so that compile doesn't change genesis. Build neard, runtime-params-estimator or test with protocol_feature_evm enabled works as before in evm-precompile. Build neard with protocol_feature_evm, genesis config doesn't change, `neard init` only add a `evm` record for convenience purpose. For existing testnet/mainnet, there's no need to hard fork, instead new protocol version (or nightly features protocol_feature_evm)  will work. Also note, in testnet and mainnet, it's required to create an evm account via link drop to make evm function call works

Test Plan
------------
- `cargo build -p neard`, `neard init` gives genesis with no evm related field and `neard run` works
- `cargo build -p neard --features protocol_feature_evm,nightly_protocol_features`, `neard init` gives no genesis config change, add a `evm` record for convinience to test purpose, and `neard run` works
- above two neard init also works with the other neard binary's neard run (backward compatible)
- standard_ops test passes
- Follow updated instruction in emu-cost/README.md to build runtime-params-estimator, then
```
cd /host/nearcore/runtime/near-evm-runner/tests
../../runtime-params-estimator/emu-cost/counter_plugin/qemu-x86_64 -cpu Westmere-v1 -plugin file=../../runtime-params-estimator/emu-cost/counter_plugin/libcounter.so ../../../target/release/runtime-params-estimator --home /tmp/data --accounts-num 200000 --iters 1 --warmup-iters 1 --evm-only
```
it still works. To speedup the test, replace 200000 with 200.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -77,8 +77,8 @@ neard = { path = "./neard"}
 testlib = { path = "./test-utils/testlib" }
 
 [profile.release]
-lto = true        # Enable full link-time optimization.
-codegen-units = 1 # Use only 1 codegen-unit to enable full optimizations.
+#lto = true        # Enable full link-time optimization.
```
Comment (bowenwang1996) [inline Cargo.toml:80]: why?

File: neard/src/runtime.rs [hunk 2]
```diff
@@ -1370,9 +1370,15 @@ impl RuntimeAdapter for NightshadeRuntime {
             && chunk_epoch_id != head_next_epoch_id)
     }
 
+    #[cfg(feature = "protocol_feature_evm")]
     fn evm_chain_id(&self) -> u128 {
         self.genesis_config.evm_chain_id
     }
+
+    #[cfg(not(feature = "protocol_feature_evm"))]
+    fn evm_chain_id(&self) -> u128 {
```
Comment (bowenwang1996) [inline neard/src/runtime.rs:1379]: Do we want to leak this part? What prevent us from only including it in the trait only if the feature is enabled? @evgenykuzyakov

File: runtime/near-runtime-fees/src/lib.rs [hunk 3]
```diff
@@ -74,10 +76,12 @@ pub struct RuntimeFeesConfig {
     pub pessimistic_gas_price_inflation_ratio: Rational,
 
     /// Describes cost of running method of evm, include deploy code and call contract function
+    #[cfg(feature = "protocol_feature_evm")]
     pub evm_config: EvmCostConfig,
 
     /// New EVM deposit.
     /// Fee to create new EVM account.
+    #[cfg(feature = "protocol_feature_evm")]
     #[serde(with = "u128_dec_format", default = "default_evm_deposit")]
```
Comment (bowenwang1996) [inline runtime/near-runtime-fees/src/lib.rs:85]: I think you can use constants in default, so there is no need for `default_evm_deposit`

File: neard/src/runtime.rs (hunk 2 above)
Comment (ailisp) [reply neard/src/runtime.rs:1379]: This is a minimum change to make evm_chain_id a noop when without protocol_feature_evm. Alternative is find all ApplyState and guard with feature there (1 definition, 7 constructings)

File: runtime/near-runtime-fees/src/lib.rs (hunk 3 above)
Comment (ailisp) [reply runtime/near-runtime-fees/src/lib.rs:85]: Doesn't work:
```
error[E0618]: expected function, found `u128`
  --> runtime/near-runtime-fees/src/lib.rs:85:49
   |
15 | const EVM_DEPOSIT: Balance = 1_000_000_000_000_000_000_000_000_000;
   | ------------------------------------------------------------------- `u128` defined here
...
57 | #[derive(Debug, Serialize, Deserialize, Clone, Hash, PartialEq, Eq)]
   |                            ----------- call expression requires function
...
85 |     #[serde(with = "u128_dec_format", default = "EVM_DEPOSIT")]
```

**Comment by evgenykuzyakov** [comment]
Looks like we're making some progress here. The tests are finally running and the sanity tests pass

**Review by artob — COMMENTED** [review]
A belated LGTM :+1: (Reviewed in full now.)

**Pull request merged**